### ISSUE TITLE
Execute powershell type provider commands in custom runspace

### ISSUE THREAD

**Issue by sergii-s** [fsprojects/FSharp.Management#79]
### Description

For powershell provider only one runspace is created for all the commands from the module. This provokes exceptions if the command is called in parallel manner.
Is there some trick or walkaround?

### Repro steps

Just run the [example ](http://fsprojects.github.io/FSharp.Management/PowerShellProvider.html) with Async.Parallel
 
```
        [1..10] 
        |> Seq.map (fun i -> async { match PS.``Get-EventLog``(logName="Application", entryType=[|"Error"|], newest=2) with
                                        | Success s -> printfn "success"
                                        | Failure s -> printfn "failure" })
        |> Async.Parallel 
        |> Async.RunSynchronously 
        |> ignore
```

### Actual behavior

Exception is thrown
System.Management.Automation.PSInvalidOperationException : The pipeline was not run because a pipeline is already running. Pipelines cannot be run concurrently.

### Expected behavior

With current implementation it's normal that it hangs. Is there a possibility to inject custom runspace to separate parallel executions ?

### Known workarounds

Searching for one

**Comment by sergey-tihon** [comment]
Yes, you are right, there is no workaround for this.

Provided method choose current runtime (`runspace`) and execute command inside.
https://github.com/fsprojects/FSharp.Management/blob/master/src/FSharp.Management.PowerShell/PowerShellProvider.fs#L65-L66

The idea was to minimize number of runspaces for cpu & memory efficiency.

It is probably too expensive to create `runspace` for each call, but we probably could create one per execution thread using [ThreadLocal](https://msdn.microsoft.com/en-us/library/dd642243(v=vs.110).aspx)

@sergii-s Would you like to implement it?

**Comment by sergii-s** [comment]
@sergey-tihon I will check the complexity. I prefer add this feature rather then use System.Management.Automation :)

About the way of implementing : i'm not a fan of threadlocal and AsyncLocal is only from 4.6. 
Optional variable "runspace" would be more elegant and flexible solution in my opinion, but I didn't check the sources yet(or a very little) so I cannot estimate the cost of this solution

**Comment by sergey-tihon** [comment]
It should not be very hard.

If you decide to go with ThreadLocal - here is the place
https://github.com/fsprojects/FSharp.Management/blob/master/src/FSharp.Management.PowerShell/Runtime.fs#L7-L21
Private `runtimes` variables should resolve `ThreadLocal< IPSRuntime >` instead of `IPSRuntime`.

In case of optional parameter (that probably will be the last one) the place is here https://github.com/fsprojects/FSharp.Management/blob/master/src/FSharp.Management.PowerShell/PowerShellProvider.fs#L65-L66 So you have to check if param is specified and use it or request one from `runtimes` map .

Note, that last option is tricky, because this `runspace` that should be initialized in the similar way lite TP do it (with all loaded modules and snapins) https://github.com/fsprojects/FSharp.Management/blob/master/src/FSharp.Management.PowerShell/Runtime.fs#L7-L21

**Comment by sergey-tihon** [comment]
Implemented in #80; released in [v0.4.4](https://www.nuget.org/packages/FSharp.Management/0.4.4)
